src/main/java/main/Parser.java

```diff
@@ -0,0 +1,184 @@
+package main;
+
+import command.ExitCommand;
+import command.FindCommand;
+import command.UndoCommand;
+import exception.DialogixException;
+import task.TaskType;
+import command.Command;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * Command
+ */
+public class Parser {
+    public static Command parse(String fullCommand) throws DialogixException {
```

Comment:
building on saltedfishxx's point, you might be able to shorten these lines of if-else statements by working through the logic you use.